Documentation/diff-options.txt

```diff
@@ -293,11 +293,13 @@ explained for the configuration variable `core.quotePath` (see
 linkgit:git-config[1]).
 
 --name-only::
-	Show only names of changed files.
+	Show only names of changed files. The output is always encoded in
+	UTF-8 if this switch is present.
 
```

Comment:
There is nothing that guarantees that "always".

As the page referred to in Peff's message says, "Note that Git at the core level treats path names simply as sequences of non-NUL bytes, there are no path name encoding conversions (except on Mac and Windows).",
it depends on platform's encoding, and hence depends on the project that recorded the history. They may have used legacy UNIX and recorded their trees in ISO-8859-1 (Latin-1), and the project's repository would work perfectly find among the participants of the project. Or they originated the project on more modern Linux and happened to have chosen UTF-8.

The same comment applies to the change in the other hunk.